### PR TITLE
Comments

### PR DESCRIPTION
**Pull request by BertZZ**
Trying to make a new Pull request to get travis to pass

### CODE REVIEW

**Comment by BertZZ** [comment]
Great, Now im getting that error that you were telling me about @henryhobhouse where my test to check that the post is going into the first p tag in the html is failing. Any fix?

**Comment by TimRobinson1** [comment]
Don't forget to keep the branch up to date with the master in case of conflicts.

**Comment by BertZZ** [comment]
Will do. Code looks good though yeah?

**Review by TimRobinson1 — APPROVED** [review]
Looks good to me.  Good edge case test with the empty comment.  I ran your tests and they're passing, just make sure you check again when you update/pull from the master before heading on.

File: spec/models/comment_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  it 'can be created with user id and post id' do
+    user = User.create(email: "test@example.com", password: "secret")
+    post = Post.create(message: "post message")
+    comment = Comment.create(body: "This is a comment", post_id: post.id, user_id: user.id)
+    expect(comment.user_id).to eq user.id
+    expect(comment.post_id).to eq post.id
+    expect(comment.body).to eq("This is a comment")
+    expect(comment.id).not_to be_nil
+  end
+  it 'Cannot be empty' do
```
Comment (neoeno) [inline spec/models/comment_spec.rb]: extra line required before here I'd say

File: spec/models/comment_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+RSpec.describe Comment, type: :model do
+  it 'can be created with user id and post id' do
+    user = User.create(email: "test@example.com", password: "secret")
+    post = Post.create(message: "post message")
+    comment = Comment.create(body: "This is a comment", post_id: post.id, user_id: user.id)
+    expect(comment.user_id).to eq user.id
+    expect(comment.post_id).to eq post.id
+    expect(comment.body).to eq("This is a comment")
+    expect(comment.id).not_to be_nil
```
Comment (neoeno) [inline spec/models/comment_spec.rb]: I think you can say this like:

```ruby
expect(comment).to be_valid
```

**Review by neoeno — CHANGES_REQUESTED** [review]
Nice, just a couple small changes.

To make this PR even better I'd add some really basic front-end functionality and a feature test for it. Doesn't have to look pretty, just shove out the comments next to each post. This 'proves' (to yourself mainly, but also to reviewers) that you've built something that will plausibly work for its intended use-case.

Also makes sure it doesn't get forgotten and live in the models doing nothing for years!

(Don't worry about making that change to this PR now though — just advice for the future)

File: spec/models/comment_spec.rb [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    expect(comment.post_id).to eq post.id
+    expect(comment.body).to eq("This is a comment")
+    expect(comment).to be_valid
+  end
+
+  it 'Cannot be empty' do
+    user = User.create(email: "test@example.com", password: "secret")
+    post = Post.create(message: "post message")
+    comment = Comment.create(body: " ", post_id: post.id, user_id: user.id)
+    expect(comment.id).to be_nil
```
Comment (neoeno) [inline spec/models/comment_spec.rb]: Sorry, I meant throughout the file! `expect(comment).to_not be_valid`

File: spec/models/comment_spec.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  it 'Cannot be longer than 150 characters' do
+    user = User.create(email: "test@example.com", password: "secret")
+    post = Post.create(message: "post message")
+    comment = Comment.create(body: "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.", post_id: post.id, user_id: user.id)
+    expect(comment.id).to be_nil
```
Comment (neoeno) [inline spec/models/comment_spec.rb]: And here

File: spec/models/comment_spec.rb [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  end
+
+  it 'Cannot be shorter than 2 characters' do
+    user = User.create(email: "test@example.com", password: "secret")
+    post = Post.create(message: "post message")
+    comment = Comment.create(body: "A", post_id: post.id, user_id: user.id)
+    expect(comment.id).to be_nil
```
Comment (neoeno) [inline spec/models/comment_spec.rb]: And here

**Review by neoeno — CHANGES_REQUESTED** [review]

**Comment by neoeno** [comment]
You've added a photo of yourself and of an owl.

**Comment by BertZZ** [comment]
Dammit!! I was code reviewing the photos form....i'll figure it out

**Review by neoeno — APPROVED** [review]

**Comment by neoeno** [comment]
Quick, merge it before anyone else merges anything!
